src/subscription_manager/cli_command/abstract_syspurpose.py

```diff
@@ -155,6 +155,17 @@ def _validate_options(self):
             else:
                 system_exit(ERR_NOT_REGISTERED_CODE, ERR_NOT_REGISTERED_MSG)
 
+        if self.is_registered() and (
```

Comment:
Can we implement this for the service-level command as well? (In a separate commit)

Reply:
Why in a separate commit and not in a single one? service-level is syspurpose subcommand, and has been left out kind of by accident, because it implements its own validation.

Reply:
It could be in the same commit if preferred.